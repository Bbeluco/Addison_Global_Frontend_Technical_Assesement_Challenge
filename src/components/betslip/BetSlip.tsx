import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { TouchableOpacity } from "react-native";
import { useMktpBetHooksType } from "../../types/useMktpBetHooksType";

interface BetSlipType {
    betOptions: useMktpBetHooksType[],
    removeItemFromArray: (itemToRemove: number) => void,
    closeDrawer: () => void
}

function BetSlip({ betOptions, removeItemFromArray, closeDrawer }: BetSlipType): React.JSX.Element {
    return (
        <View>
            <TouchableOpacity style={styles.btnCloseDrawer} onPress={() => closeDrawer()}>
                <Text style={styles.textBtnCloseDrawer}>X</Text>
            </TouchableOpacity>
            {betOptions?.map((betOption, index) => (
                <View key={index * Math.random()} style={styles.bets}>
                    <Text style={styles.betOption}>{`${betOption['teamName']} ${betOption['titleBet']}`}</Text>
                    <Text style={[styles.betPrice, styles.defaultPadding]}>{betOption['price']}</Text>
                    <TouchableOpacity 
                        style={styles.btnDeleteBet} 
                        onPress={() => removeItemFromArray(index)}
                    >
                        <Text style={styles.btnDeleteBetText}>DELETE</Text>
                    </TouchableOpacity>
                    <View style={[styles.hr, styles.defaultMargin]}/>
                </View>
            ))}
        </View>
      );
}

const styles = StyleSheet.create({
    defaultPadding: {
        padding: 10
    },
    defaultMargin: {
        margin: 10
    },
    bets: {
        marginTop: 50,
        alignItems: 'center'
    },
    betOption: {
        fontWeight: "bold"
    },
    betPrice: {
        fontWeight: "bold",
        fontSize: 20
    },
    btnDeleteBet: {
        backgroundColor: 'black',
        borderRadius: 5
    },
    btnDeleteBetText: {
        color: 'white',
        margin: 5
    },
    hr: {
        borderBottomColor: 'black',
        width: '80%',
        borderWidth: 1,
        backgroundColor: 'black'
    },
    textBtnCloseDrawer: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    btnCloseDrawer: {
        margin: 20
    }
})

export default BetSlip;