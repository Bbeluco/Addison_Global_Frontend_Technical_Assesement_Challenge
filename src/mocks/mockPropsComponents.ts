import { SEPARATOR } from "../constants/selectOptionConstants"
import mockReturnApi from "./mockReturnApi"

export const betOptionMock = [{ 'option': `button${SEPARATOR}0${SEPARATOR}0${SEPARATOR}0`, 'teamName': 'b', 'titleBet': 'c', 'price': 1 }]

export const pushSelectedOptionInfoMock = {
    add(option: string, teamName: string, titleBet: string, price: number) {},
    remove(itemRemove: number) {},
}

export const navigator = {
    onClose() {},
}

// export function pushSelectedOptionInfoMock(option: string, teamName: string, titleBet: string, price: number) {}
export const titleBetMock = 'Bet123'
export const selectionBetNameMock = 'To win'
export const selectionPriceMock = 12
export const eventIndexMock = 0
export const marketIndexMock = 0
export const selectionIndexMock = 0
export const marketNameMock = 'Market name'
export const selectionsMock = mockReturnApi[0]['markets'][0]['selections']
