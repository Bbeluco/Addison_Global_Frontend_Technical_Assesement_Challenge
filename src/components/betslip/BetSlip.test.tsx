import BetSlip from "./BetSlip";
import renderer from 'react-test-renderer';
import { betOptionMock, pushSelectedOptionInfoMock, navigator } from "../../mocks/mockPropsComponents";
import { fireEvent, render, screen } from "@testing-library/react-native";

describe('<BetSlip />', () => {
    it('should render UI correctly', () => {
        const component = renderer.create(
            <BetSlip 
                betOptions={betOptionMock}
                removeItemFromArray={pushSelectedOptionInfoMock.remove}
                closeDrawer={navigator.onClose}
            />
        ).toJSON();
        expect(component).toMatchSnapshot();
    })

    it('should call expected functions', () => {
        const spyPushSelectedOptionInfoMock = jest.spyOn(pushSelectedOptionInfoMock, 'remove');
        const spyNavigator = jest.spyOn(navigator, 'onClose')

        render(
            <BetSlip 
                betOptions={betOptionMock}
                removeItemFromArray={pushSelectedOptionInfoMock.remove}
                closeDrawer={navigator.onClose}
            />
        )

        fireEvent.press(screen.getByText('DELETE'))
        expect(spyPushSelectedOptionInfoMock).toHaveBeenCalled()
        fireEvent.press(screen.getByText('X'))
        expect(spyNavigator).toHaveBeenCalled()
    })
})