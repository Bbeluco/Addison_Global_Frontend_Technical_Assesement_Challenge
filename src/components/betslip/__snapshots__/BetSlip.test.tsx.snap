// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BetSlip /> should render UI correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "margin": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      X
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontWeight": "bold",
        }
      }
    >
      b c
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          {
            "padding": 10,
          },
        ]
      }
    >
      1
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "black",
          "borderRadius": 5,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "margin": 5,
          }
        }
      >
        DELETE
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "black",
            "borderBottomColor": "black",
            "borderWidth": 1,
            "width": "80%",
          },
          {
            "margin": 10,
          },
        ]
      }
    />
  </View>
</View>
`;
